### ISSUE TITLE
long-term goal: implement mcp-cord

### ISSUE THREAD

**Issue by nyanster** [fuzzball-muck/fuzzball#302]
The MCP website lists three packages.
https://www.moo.mud.org/mcp/index.html
It lists: mcp-negotiate, mcp-cord, and dns-org-mud-moo-simpleedit.  Only mcp-negotiate is required.  Fuzzball supports mcp-negotiate and dns-org-mud-moo-simpleedit.
Oddly, it also claims to also support org-fuzzball-simpleedit, even though in the C source code it seems that it's exactly the same as the regular dns-org-mud-moo-simpleedit: this COULD be changed if org-fuzzball-simpleedit is extended later, maybe to utilize words like mpi-code and muf-code rather than moo-code.

What isn't supported is the optional mcp-cord.  It's not urgent, but to someday implement this package would provide a tool for MCP-using developers.

I wonder how many people use MCP. :P  BUT, MCP seems like a pretty useful too I want to begin trying out.

**Comment by nyanster** [comment]
> The following cord-related MUF primitives were considered for FB6 at one time:
> 
> CORD_OPEN ( intDescr strType -- strCordID)
> CORD_SEND (strCordID strName dictData -- )
> CORD_CLOSE ( strCordID -- )
> 
> Shouldn't be too difficult to implement them and mcp-cord. But yes, long-term...

I see!
Half-relatedly, I read a useful explanation of cords and their rationale here:
https://lisdude.com/moo/mcp_mail/msg00174.html
It reminds me of how on the MUCK Hope Island, it's possible to read and write to multiple bulletin boards from any location.  (or at least, the bulletin board has sections and portrays itself as multiple :D )
That mailing list archive mostly documents the development of MCP 2.1, and I've amused myself skimming through some of it.

**Comment by revarbat** [comment]
At this point, I actually have no memory at all about if/why mcp-cords were never implemented.
I remember that org-fuzzball-simpleedit did have a minor difference from dns-org-mud-moo-simpleedit, but I think it was just a few more allowed `type` values.

**Comment by revarbat** [comment]
Verified.  The standard simpleedit accepts the following types:
- string
- string-list
- moo-code

The fuzzball variant of simpleedit accepts those and also:
- muf-code
- integer
- string

**Comment by aidanPB** [comment]
Is there a reason fuzzball simpleedit accepts `moo-code`? Is there a plan to make code written in the MOO language meaningful to FB?

**Comment by revarbat** [comment]
Yes.  Because the simpleedit MCP package was originally created for MOO systems.  Since Trebuchet is nominally server agnostic, it can accept the `moo-code` type and edits it like a `string-list`.  The fact that Trebuchet will accept a `moo-code` type simpleedit from a Fuzzball MUCK is mostly a quirky and harmless side effect.
